controllers/servingruntime_controller.go

```diff
@@ -369,23 +412,44 @@ func (r *ServingRuntimeReconciler) getRuntimesSupportingPredictor(ctx context.Co
 		return nil, err
 	}
 
+	// list all cluster serving runtimes
+	csrs := &kserveapi.ClusterServingRuntimeList{}
+	if err := r.Client.List(ctx, csrs); err != nil {
+		return nil, err
+	}
+
 	restProxyEnabled := r.ConfigProvider.GetConfig().RESTProxy.Enabled
 	srnns := make([]types.NamespacedName, 0, len(runtimes.Items))
 	for i := range runtimes.Items {
 		rt := &runtimes.Items[i]
-		if rt.Spec.IsMultiModelRuntime() && runtimeSupportsPredictor(rt, p, restProxyEnabled) {
+		if rt.Spec.IsMultiModelRuntime() && runtimeSupportsPredictor(&rt.Spec, p, restProxyEnabled, types.NamespacedName{Name: rt.Name, Namespace: rt.Namespace}) {
 			srnns = append(srnns, types.NamespacedName{
 				Name:      rt.GetName(),
 				Namespace: p.Namespace,
 			})
 		}
 	}
+	csrnns := make([]types.NamespacedName, 0, len(csrs.Items))
+	for i := range csrs.Items {
+		crt := &csrs.Items[i]
+		// Check whether a ServingRuntime exists with same name
+		if crt.Spec.IsMultiModelRuntime() && !r.runtimeExists(runtimes, crt.Name) &&
+			runtimeSupportsPredictor(&crt.Spec, p, restProxyEnabled, types.NamespacedName{Name: crt.Name}) {
+			csrnns = append(csrnns, types.NamespacedName{
+				Name:      crt.GetName(),
+				Namespace: p.Namespace,
+			})
+		}
+	}
```

Comment:
We could probably simplify the list merging a bit, e.g. by populating a map instead of a slice (then an explicit runtimeExists check won't be needed)